extract_verbs.sh

```diff
@@ -8,115 +8,140 @@ set -e
+    "mongolian" | "mn")
+        LANGUAGE="mongolian"
+        LANGCODE="mn"
+        ;;
+    "polish" | "pl")
+        LANGUAGE="polish"
+        LANGCODE="pl"
+        ;;
+    "portuguese" | "pt")
+        LANGUAGE="portuguese"
+        LANGCODE="pt"
+        ;;
+    "russian" | "ru")
+        LANGUAGE="russian"
+        LANGCODE="ru"
+        ;;
+    "slovak" | "sk")
+        LANGUAGE="slovak"
+        LANGCODE="sk"
+        ;;
+    "spanish" | "es")
+        LANGUAGE="spanish"
+        LANGCODE="es"
+        ;;
+    "swahili" | "sw")
+        LANGUAGE="swahili"
+        LANGCODE="sw"
+        ;;
+    *)
+        echo "invalid language parameter: quitting"
+        exit 1
+        ;;
 esac
 
-# Form Wikipedia dump URL
+# download Wikipedia dump
 URL="http://download.wikimedia.org/"$LANGCODE"wiki/latest/"$LANGCODE"wiki-latest-pages-articles.xml.bz2"
-echo "Downloading dump from: $URL"
+echo "[CHECK] downloading Wikipedia dump to " $(pwd)
 wget $URL
-# Extract text
-if [ ! -d "extracted"]; then
+echo "[CHECK] downloaded Wikipedia dump to " $(pwd)
+
+# create directory to extract text
+if [ ! -d "extracted" ]; then
     mkdir extracted
+    echo "[CHECK] extracted directory created in " $(pwd)
+else
+    echo "[CHECK] extracted directory already exists in " $(pwd)
 fi
-bzcat "$LANGCODE"wiki-latest-pages-articles.xml.bz2 | scripts/lib/WikiExtractor.py -o extracted
-# Split extraction by article
-if [ ! -d "corpus"]; then
+
+# extract files (note: check system path to fact-extractor directory)
+bzcat "$LANGCODE"wiki-latest-pages-articles.xml.bz2 | fact-extractor/lib/WikiExtractor.py -o extracted
+echo "[CHECK] extracted files"
+
+# create directory to split extraction by article
+if [ ! -d "corpus" ]; then
     mkdir corpus
+    echo "[CHECK] corpus directory created in " $(pwd)
+else
+    echo "[CHECK] corpus directory already exists in " $(pwd)
 fi
-cat extracted/*/* | csplit --suppress-matched -z -f 'corpus/doc_' - '/</doc>/' {*}
+
+# split files and remove headers
+cat extracted/*/* | sed 's/<doc id="[[:alnum:]]*" url="[[:alnum:]\:\/\.\?\=]*" title="[[:alnum:][:space:]]*">//g' | csplit -z -s -f 'corpus/doc_' - '/</doc>/' {*}
+sed -i 's/<\/doc>//g' corpus/*
+echo "[CHECK] split files"
+
+# cat extracted/*/* | csplit --suppress-matched -z -f 'corpus/doc_' - '/</doc>/' {*}
 # Build a single big file
-find extracted -type f -exec cat {} \; > all-extracted.txt
+# find extracted -type f -exec cat {} \; > all-extracted.txt
 # Extract verbs with TreeTagger
 # N.B. treetagger segfaults with the single big file, run it over each article instead
-#cat all-extracted.txt | treetagger/cmd/tree-tagger-italian | grep VER | sort -u > verbi.txt
-find extracted -type f -exec bash -c "cat '{}' | treetagger/cmd/tree-tagger-"$LANGUAGE" | grep VER >> verbs" \;
+
+# extract verbs and sort (note: check system path to tree-tagger directory)
+find corpus -type f -exec bash -c "cat '{}' | tree-tagger/cmd/tree-tagger-"$LANGUAGE" | grep V[[:upper:]*] >> verbs" \;
```

Comment:
agreed. after going through the tagsets of several languages, the only commonality I found for verbs (and only verbs unless Im mistaken) was the 'V' at the beginning. would you suggest keeping language specific verb tagsets in the script?